### PR TITLE
10288 training data version warning

### PR DESCRIPTION
**Pull request by carlad**
**Proposed changes**:
Improvements based on @ancalita's comments on https://github.com/RasaHQ/rasa/pull/10352

### CODE REVIEW

File: rasa/shared/utils/validation.py [hunk 1]
```diff
@@ -241,24 +241,25 @@ def validate_training_data_format_version(
         )
         return True
 
+    if str(version_value) < LATEST_TRAINING_DATA_FORMAT_VERSION:
```
Comment (ancalita) [inline rasa/shared/utils/validation.py:244]: I think this should actually come after L260, just before L261 `if`, to prioritise raising the `TypeError` if a legacy version is being.

**Review by ancalita — COMMENTED** [review]
Thank you for looking into the suggestions 🙏🏼

File: rasa/shared/utils/validation.py [hunk 2]
```diff
@@ -246,19 +246,20 @@ def validate_training_data_format_version(
...
+        if str(version_value) < LATEST_TRAINING_DATA_FORMAT_VERSION:
+            rasa.shared.utils.io.raise_warning(
+                f"Training data file {filename} has a lower "
+                f"format version than your Rasa Open Source installation: "
+                f"{version_value} < {LATEST_TRAINING_DATA_FORMAT_VERSION}. "
+                f"Rasa Open Source will read the file as a version "
+                f"{LATEST_TRAINING_DATA_FORMAT_VERSION} file. "
+                f"Please update your version key to "
+                f"{LATEST_TRAINING_DATA_FORMAT_VERSION}. "
+                f"See {DOCS_URL_TRAINING_DATA}."
```
Comment (ancalita) [inline rasa/shared/utils/validation.py:258]: Just noticed this docs link reference, could you have a look to see if we need to insert a comment that the training data version in `3.0` should be updated to `3.0`?

Comment (ancalita) [reply rasa/shared/utils/validation.py:258]: in that docs page, I mean, sorry keep pressing enter without finishing my comment 😅

**Review by ancalita — APPROVED** [review]
🎉

File: rasa/shared/utils/validation.py [hunk 3]
```diff
@@ -246,19 +246,20 @@ def validate_training_data_format_version(
         if isinstance(parsed_version, LegacyVersion):
             raise TypeError
 
+        if str(version_value) < LATEST_TRAINING_DATA_FORMAT_VERSION:
```
Comment (joejuzl) [inline rasa/shared/utils/validation.py:249]: Shouldn't we be using `version.parse` for this check?

Comment (carlad) [reply rasa/shared/utils/validation.py:249]: we can actually use `parsed_version` (which is `version.parse(version_value)`)

Comment (carlad) [reply rasa/shared/utils/validation.py:249]: PR: https://github.com/RasaHQ/rasa/pull/10366

**Pull request merged**